homeassistant/components/microbees/switch.py

```diff
@@ -19,13 +19,13 @@ async def async_setup_entry(
 ) -> None:
     """Config entry."""
     coordinator = hass.data[DOMAIN][entry.entry_id].coordinator
-    switches = []
-    for bee_id, bee in coordinator.data.bees.items():
-        if bee.productID in (25, 26, 27, 35, 38, 46, 63, 64, 65, 86):
-            for switch in bee.actuators:
-                switches.append(MBSwitch(coordinator, bee_id, switch.id))
 
-    async_add_entities(switches)
+    async_add_entities(
+        MBSwitch(coordinator, bee_id, switch.id)
+        for bee_id, bee in coordinator.data.bees.items()
+        if bee.productID in (25, 26, 27, 35, 38, 46, 63, 64, 65, 86)
```

Comment:
`x in s` has an average time complexity of `O(n)` here. https://wiki.python.org/moin/TimeComplexity

If you made `(25, 26, 27, 35, 38, 46, 63, 64, 65, 86)` a constant `set` it would have average time complexity of `O(1)`

Reply:
Yes, I commented on the other PR that the entire integration should probably use an IntEnum for those integers to make it more pythonic.  Then you could make it a set.

Since I have no idea what those ID's are, I can't do it myself.

Reply:
```diff
diff --git a/homeassistant/components/microbees/switch.py b/homeassistant/components/microbees/switch.py
index 33db881932e..fcc83670efe 100644
--- a/homeassistant/components/microbees/switch.py
+++ b/homeassistant/components/microbees/switch.py
@@ -12,7 +12,7 @@ from .coordinator import MicroBeesUpdateCoordinator
 from .entity import MicroBeesActuatorEntity
 
 SOCKET_TRANSLATIONS = {46: "socket_it", 38: "socket_eu"}
-
+SWITCH_PRODUCT_IDS = {25, 26, 27, 35, 38, 46, 63, 64, 65, 86}
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
@@ -21,7 +21,7 @@ async def async_setup_entry(
     coordinator = hass.data[DOMAIN][entry.entry_id].coordinator
     switches = []
     for bee_id, bee in coordinator.data.bees.items():
-        if bee.productID in (25, 26, 27, 35, 38, 46, 63, 64, 65, 86):
+        if bee.productID in SWITCH_PRODUCT_IDS:
             for switch in bee.actuators:
                 switches.append(MBSwitch(coordinator, bee_id, switch.id))
 
```

Reply:
Something like the above (note: code based on `dev`)